rust/crd/src/lib.rs

```diff
@@ -171,8 +171,9 @@ pub struct SupersetClusterSpec {
 #[derive(Clone, Debug, Deserialize, JsonSchema, PartialEq, Serialize)]
 #[serde(rename_all = "camelCase")]
 pub struct SupersetClusterConfig {
-    /// Mandatory list of AuthenticationClasses used to authenticate users.
-    pub authentication: Vec<SupersetClientAuthenticationDetails>,
+    /// List of AuthenticationClasses used to authenticate users.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub authentication: Option<Vec<SupersetClientAuthenticationDetails>>,
```

Comment:
Can you use Superset without any authentication at all? And if yes, I would prefer `[serde(default)]` over  ` Option<Vec<_>>`

Reply:
Ah I see. I though the DB authentication could go in here as well (static AuthClass), but it's hard-coded in some Secret as of now